{"kind":"symbol","variants":[{"paths":["\/documentation\/amplify\/amplifytask\/failure"],"traits":[{"interfaceLanguage":"swift"}]}],"sections":[],"schemaVersion":{"minor":3,"major":0,"patch":0},"hierarchy":{"paths":[["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AmplifyTask"]]},"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AmplifyTask\/Failure"},"metadata":{"required":true,"role":"symbol","title":"Failure","roleHeading":"Associated Type","modules":[{"name":"Amplify"}],"symbolKind":"associatedtype","fragments":[{"kind":"keyword","text":"associatedtype"},{"kind":"text","text":" "},{"kind":"identifier","text":"Failure"},{"kind":"text","text":" : "},{"preciseIdentifier":"s:7Amplify0A5ErrorP","text":"AmplifyError","kind":"typeIdentifier"}],"externalID":"s:7Amplify0A4TaskP7FailureQa"},"primaryContentSections":[{"declarations":[{"languages":["swift"],"tokens":[{"text":"associatedtype","kind":"keyword"},{"text":" ","kind":"text"},{"text":"Failure","kind":"identifier"},{"text":" : ","kind":"text"},{"text":"AmplifyError","preciseIdentifier":"s:7Amplify0A5ErrorP","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AmplifyError","kind":"typeIdentifier"}],"platforms":["iOS"]}],"kind":"declarations"}],"references":{"doc://amplify-swift.Amplify/documentation/Amplify/AmplifyTask/Failure":{"fragments":[{"text":"associatedtype","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"Failure"},{"kind":"text","text":" : "},{"preciseIdentifier":"s:7Amplify0A5ErrorP","kind":"typeIdentifier","text":"AmplifyError"}],"title":"Failure","kind":"symbol","type":"topic","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AmplifyTask\/Failure","required":true,"url":"\/documentation\/amplify\/amplifytask\/failure","abstract":[],"role":"symbol"},"doc://amplify-swift.Amplify/documentation/Amplify":{"url":"\/documentation\/amplify","type":"topic","title":"Amplify","role":"collection","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify","kind":"symbol","abstract":[]},"doc://amplify-swift.Amplify/documentation/Amplify/AmplifyTask":{"role":"symbol","abstract":[{"text":"Simple task that represents a unit of work and its result.","type":"text"}],"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AmplifyTask","title":"AmplifyTask","url":"\/documentation\/amplify\/amplifytask","type":"topic","fragments":[{"kind":"keyword","text":"protocol"},{"text":" ","kind":"text"},{"text":"AmplifyTask","kind":"identifier"}],"navigatorTitle":[{"kind":"identifier","text":"AmplifyTask"}],"kind":"symbol"},"doc://amplify-swift.Amplify/documentation/Amplify/AmplifyError":{"navigatorTitle":[{"kind":"identifier","text":"AmplifyError"}],"role":"symbol","title":"AmplifyError","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AmplifyError","abstract":[{"type":"text","text":"Amplifyâ€™s philosophy is to expose friendly error messages to the customer that assist with debugging."},{"type":"text","text":" "},{"type":"text","text":"Therefore, failable APIs are declared to return error results with Amplify errors, which require"},{"type":"text","text":" "},{"type":"text","text":"recovery suggestions and error messages."}],"kind":"symbol","type":"topic","fragments":[{"text":"protocol","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"AmplifyError"}],"url":"\/documentation\/amplify\/amplifyerror"}}}